src/endpoint/s3/s3_rest.js

```diff
@@ -364,7 +364,7 @@ function handle_error(req, res, err) {
         ((err instanceof S3Error) && err) ||
         new S3Error(S3Error.RPC_ERRORS_TO_S3[err.rpc_code] || S3Error.InternalError);
 
-    if (s3err.code === 'MalformedPolicy') {
+    if (!(err instanceof S3Error) && s3err.code === 'MalformedPolicy') {
```

Comment:
is it needed?

Reply:
yes, otherwise it will fail for s3 errors because rpc_data is undefined. seems like the original intention was that MalformedPolicy will always be thrown as a RpcError and that the message and detail will be overwritten by its values